docs-src/ref-programs-compute.scrbl

```diff
@@ -892,6 +902,16 @@ This may be abbreviated as @reachin{arr.reduce(z, f)}.
 This function is generalized to an arbitrary number of arrays of the same size, which are provided before the @reachin{z} argument.
 For example, @reachin{Array.iota(4).reduce(Array.iota(4), 0, (x, y, z) => (z + x + y))} returns @reachin{((((0 + 0 + 0) + 1 + 1) + 2 + 2) + 3 + 3)}.
 
+@subsubsection{@tt{Array.reduceWithIndex} && @tt{.reduceWithIndex}}
+
+@(mint-define! '("reduceWithIndex"))
+@reach{
+  Array.reduceWithIndex(arr, z, f)
+  arr.reduceWithIndex(z, f) }
+
+@index{Array.reduceWithIndex} @reachin{Array.reduceWithIndex(arr, z, f)} is the same as @reachin{Array.reduce}, except it
+provides @reachin{f} with an additional argument, which is the index of the current element in @reachin{arr}.
```

Comment:
This, and `mapWithIndex` are not exactly alike, because I believe that those functions take N arguments and do implicit zipping. I think these functions should mention that they only work on one argument and if you want the indices, then use `withIndex` first on one of the arrays